app/src/main/java/org/jellyfin/androidtv/data/repository/NetworkStatusRepository.kt

```diff
@@ -0,0 +1,86 @@
+package org.jellyfin.androidtv.data.repository
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.update
+import timber.log.Timber
+
+
+enum class NetworkState {
+	CONNECTED,
+	DISCONNECTED,
+	UNKNOWN
+}
+
+interface NetworkStatusRepository {
+	val state: StateFlow<NetworkState>
+}
+
+/**
+ * Registers itself to monitor the application network status
+ * and provides a StateFlow to monitor the network status
+ */
+class NetworkStatusRepositoryImpl(
+	appContext: Context,
+	appScope: CoroutineScope,
+	// DI injection to make repo instantiation much easier in tests:
+	connectivityManagerDi: ConnectivityManager? = null,
+	private val networkRequestBuilderDi: NetworkRequest.Builder? = null
+) : NetworkStatusRepository {
+
+	private val _state = MutableStateFlow(NetworkState.UNKNOWN)
+	override val state = _state.asStateFlow()
+
+	// Some manual DI, since using Koin here is overkill
+	private val connectivityManager = connectivityManagerDi
+		?: appContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
```

Comment:
Add a `factory<ConnectivityManager> { applicationContexT().getSystemService(Context.CONNECTIVITY_SERVICE) }` to Koin so we can just use the service via DI.